src/v/cluster/self_test_backend.cc

```diff
@@ -51,6 +52,25 @@ ss::future<> self_test_backend::stop() {
     co_await std::move(f);
 }
 
+void self_test_backend::parse_unknown_checks(
+  std::vector<cloudcheck_opts>& ctos,
+  std::vector<unknown_check>& unknown_checks) {
+    for (auto it = unknown_checks.begin(); it != unknown_checks.end();) {
+        if (it->test_type == "cloud") {
+            json::Document doc;
+            if (doc.Parse(it->test_json.c_str()).HasParseError()) {
+                ++it;
+                continue;
+            }
+            const auto& obj = doc.GetObject();
+            ctos.push_back(cluster::cloudcheck_opts::from_json(obj));
+            it = unknown_checks.erase(it);
```

Comment:
Can we pull out some of the logic in admin/server.cc that parses known checks and reuse that here? It'll be really easy otherwise to forget to update this list the next time a check is added

Reply:
I considered this, but due to how we are operating directly on the `std::vector<>` of unknown opts here, it becomes a little bit awkward to try to abstract both sections of code into a generic function.

I agree it may be easy to forget to update this list, and noted as such in the declaration comment, but given the function signatures do you think this is okay?

Reply:
Refactored some of the things we discussed into `parse_self_test_checks()`.